core/src/com/unciv/logic/civilization/CityStateFunctions.kt

```diff
@@ -667,13 +668,10 @@ class CityStateFunctions(val civInfo: CivilizationInfo) {
         }
     }
 
-    fun getCityStateResourcesForAlly(): ResourceSupplyList {
-        val newDetailedCivResources = ResourceSupplyList()
+    fun getCityStateResourcesForAlly() = ResourceSupplyList().apply {
```

Comment:
Idiomatic Trog maybe, maybe kotlin best practice - who am I to know. Will try to keep an eye on excessive use. There's probably even some I may have perpetrated using a `this@` to get at the shadowed 'main' `this` and that's another heavy factor against the construct. Can save a **lot** of text though.

Reply:
OK, after looking at source in Studio, you're definitely right. This construct mimics the builder pattern, and its main drawback is shadowing of an outer `this` where, if you're in a major outer class, you need to emphasize you're building something else...

Brainstorm: `new().let { newObject ->` would also give the 'builder' feeling while not shadowing `this`... But I hate `let`... ??? This or revert that construct in 3 of 4 places.

Reply:
Yea yea gottitt just brain-schtorming... how to avoid 
```kotlin
fun fgh() {
    val newSausage = Sausage()
    newSausage.buy()
    newSausage.bringHome()
    newSausage.findKeys()
    newSausage.askNeigborForKeys()
    newSausage.smashWindow()
    newSausage.goKitchen()
    newSausage.openMicrowave()
    ....
    return newSausage
}
```

Reply:
Too many Sausages for the Vegan! Why else? (Imagine 20 of those cheap things from a tin can heated in a microwave. At 9 in the morning in a company kitchen. "A real man needs a healthy breakfast". Really happened).